### DISCUSSION TITLE
Support with Discord Game SDK

### DISCUSSION THREAD

**Discussion by blinkuz** [AscensionGameDev/Intersect-Engine#2384] [Ideas]
I would like to know if it would be generic and reusable enough to accept an integration with the Discord Game SDK in the engine, the idea is that it is configurable from the server and in a first implementation only the Rich Presence would be used along with its own configurations to edit the component.

Developers would only have to create the app in the discord panel and save their credentials in the server configuration.

I am open to ideas about what would be the most important elements to show in the component, by default discord brings support to show the information of the groups but there are other more open fields as you can see in the following image.

![image](https://github.com/user-attachments/assets/25682622-e81c-44d6-afac-b6605c9121bd)

**Comment by WeylonSantana** [comment]
this is a perfect and great idea for all devs
Could you show the map, the character and the level, very cool indeed

**Comment by blinkuz** [comment]
The following images show the first implementation of Discord Rich Presence using the engine's plugin system. This is the first version, where it only works client-side for now, since the logic to send configuration and sensitive information from the server still needs to be integrated.

Among the 'limitations' I’ve found is that the main description or the details field in the Rich Presence doesn't allow line breaks, and there is a character limit for the text.

In the images, you can see some basic information retrieved from the player, but I'd like to know what data you'd like to configure to display from your games, since there will be a JSON file where you can configure the text with variables. I'll upload this same information to GitHub to keep it updated.

![image](https://github.com/user-attachments/assets/f2d58c41-f5ae-44ed-b032-c42343502383)

![image](https://github.com/user-attachments/assets/582da912-ad1d-4e6d-af9c-57b328c1d0f2)

You can find the plugin code in this branch of my fork. IT'S NOT READY YET, as it only works on the client side for now. Once I finish development, I plan to upload it to its own repository, including both the server-side and client-side plugin code.

https://github.com/blinkuz/Intersect-Engine/tree/feature/add-discord-rpc-plugin

**Comment by blinkuz** [comment]
I’m closing this discussion since we talked on Discord about turning it into an external plugin rather than adding it directly to the engine. You can find the compiled version of the plugin here: https://blinkuz.itch.io/discord-rich-presence-plugin
